pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -452,94 +455,148 @@ func (g *gatewayMonitor) createGlobalNetMarkingChain() error {
 	}
 
-	forwardToSubGlobalNetChain = []string{"-j", constants.SmGlobalnetEgressChainForHeadlessSvcPods}
-	if err := g.ipt.InsertUnique("nat", constants.SmGlobalnetEgressChain, 3, forwardToSubGlobalNetChain); err != nil {
-		logger.Errorf(err, "Error inserting iptables rule %q", strings.Join(forwardToSubGlobalNetChain, " "))
+	ruleSpec = packetfilter.Rule{
+		TargetChain: constants.SmGlobalnetEgressChainForPods,
+		Action:      packetfilter.RuleActionJump,
+	}
+	if err := g.pFilter.InsertUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChain, 2, &ruleSpec); err != nil {
+		logger.Errorf(err, "Error inserting packetfilter rule %+v", ruleSpec)
 	}
 
-	forwardToSubGlobalNetChain = []string{"-j", constants.SmGlobalnetEgressChainForHeadlessSvcEPs}
-	if err := g.ipt.InsertUnique("nat", constants.SmGlobalnetEgressChain, 4, forwardToSubGlobalNetChain); err != nil {
-		logger.Errorf(err, "Error inserting iptables rule %q", strings.Join(forwardToSubGlobalNetChain, " "))
+	ruleSpec = packetfilter.Rule{
+		TargetChain: constants.SmGlobalnetEgressChainForHeadlessSvcPods,
+		Action:      packetfilter.RuleActionJump,
+	}
+	if err := g.pFilter.InsertUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChain, 3, &ruleSpec); err != nil {
+		logger.Errorf(err, "Error inserting packetfilter rule %+v", ruleSpec)
 	}
 
-	forwardToSubGlobalNetChain = []string{"-j", constants.SmGlobalnetEgressChainForNamespace}
-	if err := g.ipt.InsertUnique("nat", constants.SmGlobalnetEgressChain, 5, forwardToSubGlobalNetChain); err != nil {
-		logger.Errorf(err, "Error inserting iptables rule %q", strings.Join(forwardToSubGlobalNetChain, " "))
+	ruleSpec = packetfilter.Rule{
+		TargetChain: constants.SmGlobalnetEgressChainForHeadlessSvcEPs,
+		Action:      packetfilter.RuleActionJump,
+	}
+	if err := g.pFilter.InsertUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChain, 4, &ruleSpec); err != nil {
+		logger.Errorf(err, "Error inserting packetfilter rule %+v", ruleSpec)
+	}
+
+	ruleSpec = packetfilter.Rule{
+		TargetChain: constants.SmGlobalnetEgressChainForNamespace,
+		Action:      packetfilter.RuleActionJump,
+	}
+	if err := g.pFilter.InsertUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChain, 5, &ruleSpec); err != nil {
+		logger.Errorf(err, "Error inserting packetfilter rule %+v", ruleSpec)
 	}
 
-	forwardToSubGlobalNetChain = []string{"-j", constants.SmGlobalnetEgressChainForCluster}
-	if err := g.ipt.InsertUnique("nat", constants.SmGlobalnetEgressChain, 6, forwardToSubGlobalNetChain); err != nil {
-		logger.Errorf(err, "Error inserting iptables rule %q", strings.Join(forwardToSubGlobalNetChain, " "))
+	ruleSpec = packetfilter.Rule{
+		TargetChain: constants.SmGlobalnetEgressChainForCluster,
+		Action:      packetfilter.RuleActionJump,
+	}
+	if err := g.pFilter.InsertUnique(packetfilter.TableTypeNAT, constants.SmGlobalnetEgressChain, 6, &ruleSpec); err != nil {
```

Comment:
Just to confirm, are you planning to use the same type of chains/flow (as used in IPtables approach) even with Nftables?

Reply:
Yep, for the regular/base chains and rules, 
IPHook chains should be used for ipstack hook chains.
As per tables not sure which tables we'll use for nftables (this is why we use tabletype in new API )